### DISCUSSION TITLE
Dispose hook on widget rebuild

### DISCUSSION THREAD

**Discussion by kennethjyang** [rrousselGit/flutter_hooks#380] [Q&A] [answered]
Do hooks dispose of themselves when a widget rebuilds? Is it possible to cause this behavior?

I'm using Riverpod for state management and I have a widget that creates a custom `TextEditingController` hook in the build method. When I update my widget's state (in Riverpod) the widget correctly rebuilds but it seems to retain the old `TextEditingController` hook. Is it possible to automatically dispose of and recreate it?

Thanks!

**Comment by rrousselGit** [comment, marked as answer]
Change keys on the hook

**Reply by kennethjyang** [reply]
Assigning a key worked. Thank you!
